ci/snapcraft.sh

```diff
@@ -0,0 +1,135 @@
+#!/bin/sh
+
+set -e
+
+# This script encapsulates the job of crafting snaps of rustup and publishing
+# them.  We construct either an edge snap (builds on PRs or master)
+# or a beta snap if built from a non-master branch.
+
+# If built on master, we will publish the edge snap
+# If built on stable, we will publish the beta snap
+
+# To parameterise the build, the following environment variables are needed:
+# DO_SNAP=1  <-- without this, we don't run
+# SNAP_EDGE=1 <-- with this, we build an edge/devel snap, otherwise a release snap
+# SNAP_ARCH=... <-- without this we cannot built, this is the snap architecture we're building
+
+if ! [ -n "$DO_SNAP" ]; then
+  echo "SNAP: Not attempting to snap"
+  exit 0
+fi
+
+if ! hash snapcraft >/dev/null 2>/dev/null; then
+  echo "SNAP: Unable to find snapcraft, cannot run"
+  exit 1
+fi
+
+if ! [ -n "$SNAP_ARCH" ]; then
```

Comment:
Try `[ -z ... ]` instead.

Reply:
I have no words for why I didn't use `-z`  Fixed.